src/metabase/db/custom_migrations.clj

```diff
@@ -245,3 +245,44 @@
                                                  :where  [:and
                                                           [:<> :result_metadata nil]
                                                           [:<> :result_metadata "[]"]]})))))
+
+(defn- remove-join-alias
+  "Removes the join-alias key from the `field_ref` opts map. If the map becomes empty, replace it with nil."
+  [field_ref]
+  (match field_ref
+    ["field" id opts] ["field" id (not-empty (dissoc opts "join-alias"))]
+    _ field_ref))
+
+(defn- add-join-alias-to-visualization-field-refs [{:keys [visualization_settings result_metadata]}]
+  (let [result_metadata        (json/parse-string result_metadata)
+        visualization_settings (json/parse-string visualization_settings)
+                               ;; Previously the FE removed the join-alias from result_metadata field_refs to match against the column_settings keys
+                               ;; Now we do it here to preserve the same behaviour
+        column-key->metadata   (group-by (comp remove-join-alias #(get % "field_ref")) result_metadata)]
+    (json/generate-string
+     (update visualization_settings "column_settings"
+             (fn [column_settings]
+               (into {}
+                     (mapcat (fn [[k v]]
+                               (match (vec (json/parse-string k))
+                                 ["ref" ["field" id opts]]
+                                 (for [column-metadata (column-key->metadata ["field" id opts])]
+                                   [(json/generate-string ["ref" (get column-metadata "field_ref")]) v])
```

Comment:
This is the key line where we create new `column_settings` keys out of matching field_refs in `result_metadata`.